### PR TITLE
Add Genshin Impact game version 2.0

### PR DESCRIPTION
**Pull request by Sculas**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

Wasn't entirely sure if I should use the game version or the date (20210525204658 currently) from the downloaded file, but I went with the game version which is 2.0.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/20316)

### CODE REVIEW

**Comment by Lucaskyy** [comment]
It works fine on my PC, it gives a warning because there's no silent installer (there's none sadly and I can't decompile the installer) but it works fine besides that?

**Comment by Lucaskyy** [comment]
Alright, figured it out. The generator is outdated, and there's no message whatsoever telling that. Drafting this PR so I can redo this tomorrow with winget-create.

File: manifests/m/miHoYo/GenshinImpact/2.0.0.0.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+LicenseUrl: https://genshin.mihoyo.com/en/company/terms
+Description: "Genshin Impact is an action role-playing game developed and published by miHoYo. It was released for Microsoft Windows, PlayStation 4, Android, and iOS in September 2020, for PlayStation 5 in April 2021, and is planned for release on Nintendo Switch. The game features an open world environment and action-based battle system using elemental magic and character-switching. Since its release the game has been free-to-play and is monetized through gacha game mechanics where players can obtain new characters, weapons, and other resources."
+InstallModes:
+  - interactive
+Installers:
+  - InstallerLocale: en-US
+    Architecture: x64
+    InstallerType: EXE
+    InstallerUrl: https://ys-api-os.mihoyo.com/event/download_porter/link/ys_global/genshinimpactpc/default
+    InstallerSha256: DD16F5A4ADE5AE70B929C9360858AA13DF8FA72BF9BEB437DBA7B27CCE1E1407
```
Comment (vedantmgoyal2009) [inline manifests/m/miHoYo/GenshinImpact/2.0.0.0.yaml:17]:
```suggestion
    InstallerSha256: DD16F5A4ADE5AE70B929C9360858AA13DF8FA72BF9BEB437DBA7B27CCE1E1407
ManifestType: singleton
ManifestVersion: 1.0.0
```

**Review by vedantmgoyal9 — CHANGES_REQUESTED** [review]

File: manifests/m/miHoYo/GenshinImpact/2.0.0.0.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+PackageIdentifier: miHoYo.GenshinImpact
+Publisher: miHoYo
+PublisherUrl: https://genshin.mihoyo.com/en/home
+PackageName: Genshin Impact
+PackageVersion: 2.0.0.0
+License: Proprietary
+InstallerType: EXE
+LicenseUrl: https://genshin.mihoyo.com/en/company/terms
+Description: "Genshin Impact is an action role-playing game developed and published by miHoYo. It was released for Microsoft Windows, PlayStation 4, Android, and iOS in September 2020, for PlayStation 5 in April 2021, and is planned for release on Nintendo Switch. The game features an open world environment and action-based battle system using elemental magic and character-switching. Since its release the game has been free-to-play and is monetized through gacha game mechanics where players can obtain new characters, weapons, and other resources."
```
Comment (vedantmgoyal2009) [inline manifests/m/miHoYo/GenshinImpact/2.0.0.0.yaml:9]:
```suggestion
InstallerType: EXE
LicenseUrl: https://genshin.mihoyo.com/en/company/terms
ShortDescription: "Genshin Impact is an action role-playing game developed and published by miHoYo."
```

**Review by vedantmgoyal9 — CHANGES_REQUESTED** [review]

**Review by vedantmgoyal9 — CHANGES_REQUESTED** [review]
Change manifests/m/miHoYo/GenshinImpact/2.0.0.0.yaml to manifests/m/miHoYo/GenshinImpact/2.0.0.0/2.0.0.0.yaml

**Comment by KaranKad** [comment]
Support for interactive installation is yet to come.

**Comment by Lucaskyy** [comment]
> Support for interactive installation is yet to come.

All this currently does is it downloads the installer from Genshin Impact and then start it.
Should I close this PR, or keep it open?

**Comment by Lucaskyy** [comment]
This is indeed not possible yet, closing this PR.

**Pull request closed without merging**